Document for Getting started with service binding (#37088)
openshift/openshift-docs: modules/sbo-connecting-spring-petclinic-sample-app-to-postgresql-database-service.adoc

= Connecting the Spring PetClinic sample application to the PostgreSQL database service
[id="sbo-connecting-spring-petclinic-sample-app-to-postgresql-database-service_{context}"]

> For connecting the sample application to the database service, you have to create a `ServiceBinding` custom resource (CR) that triggers the {servicebinding-title} to project the binding data into the application.

Reviewer: ```suggestion
To connect the sample application to the database service, create a `ServiceBinding` custom resource (CR) that triggers the {servicebinding-title} to project the binding data into the application.
```
Grammatical usage: "To connect" rather than "For connecting"

Also, per IBM usage, consider either changing the main clause to the imperative our using "you must." See the "have to" entry in the _IBM Style Guide_.

[discrete]
.Procedure

. Create a `ServiceBinding` CR to project the binding data:
+
----
$ oc apply -f - << EOD
---
apiVersion: binding.operators.coreos.com/v1alpha1
kind: ServiceBinding
metadata:
    name: spring-petclinic-rest
    namespace: my-postgresql
spec:
    services: <1>
    - group: postgres-operator.crunchydata.com
      version: v1beta1
      kind: PostgresCluster <2>
      name: hippo
    - group: ""
      version: v1
      kind: Secret
      name: hippo-pguser-hippo
    application: <3>
      name: spring-petclinic-rest
      group: apps
      version: v1
      resource: deployments
----
<1> Specifies a list of service resources.
<2> The CR of the database.
<3> The sample application that points to a Deployment or any other similar resource with an embedded PodSpec.
+
The output verifies that the `ServiceBinding` CR is created to project the binding data into the sample application.
+
.Example output
----
servicebinding.binding.operators.coreos.com/spring-petclinic-rest created
----

> . To verify whether the binding is successful, check the status conditions of the binding resource:

Reviewer: ```suggestion
. To verify if the binding is successful, check the status conditions of the binding resource:
```
Per IBM Style Guide, "Use 'if' to express a condition and 'whether' to express an alternative." See "if versus whether conjunction" section

+
----
$ oc get servicebindings spring-petclinic-rest -n my-postgresql -o jsonpath-as-json='{.status.conditions}'
----
+
.Example output
----
[
  [
    {
      "lastTransitionTime": "2021-09-06T13:42:28Z",
      "message": "",
      "reason": "DataCollected",
      "status": "True",
      "type": "CollectionReady",
    },
    {
      "lastTransitionTime": "2021-09-06T13:42:28Z",
      "message": "",
      "reason": "ApplicationUpdated",
      "status": "True",
      "type": "InjectionReady",
    },
    {
      "lastTransitionTime": "2021-09-06T13:42:28Z",
      "message": "",
      "reason": "ApplicationsBound",
      "status": "True",
      "type": "Ready",
    },
  ],
]
----
+
The values from the binding data of the database service are projected as files, by default, into the workload container that runs the sample application.

> . To verify whether the files in the application contain the projected binding data, use the following command in shell to print out the contents:

Reviewer: ```suggestion
. To verify that the files in the application contain the projected binding data, use the following command in shell to print out the contents:
```
